### PR TITLE
Added Copy Message Link to context menu actions

### LINKED ISSUE #1816: Copy link to message
It would be nice if we could copy a link to a message, especially on the ContextChatViewController after searching

### LINKED ISSUE #1818: Open message links in ContextChatViewController
When a link to a message is posted that is not in the view (same for quotes) we currently do nothing. Since we have the ContextChatViewController, we could open that link there and get the context (as we do in the SharedMediaList)

### PR DESCRIPTION
**Pull request by rapterjet2004**
- fixes https://github.com/nextcloud/talk-ios/issues/1816

Screenshot 2024-10-21 at 9 29 34 AM

### CODE REVIEW

File: NextcloudTalk/BaseChatViewController.swift [hunk 1]
```diff
@@ -964,6 +964,14 @@ import QuickLook
         NotificationPresenter.shared().present(text: NSLocalizedString("Message copied", comment: ""), dismissAfterDelay: 5.0, includedStyle: .dark)
     }
 
+    func didPressCopyLink(for message: NCChatMessage) {
+        let account = NCDatabaseManager.sharedInstance().activeAccount()
+        let url = "\(account.server)/call/\(room.token!)#message_\(message.messageId)"
```
Comment (SystemKeeper) [inline NextcloudTalk/BaseChatViewController.swift]: You need to take servers into account, that do not have pretty URLs enabled. See https://github.com/nextcloud/talk-ios/blob/ab0ee8e036864c349058c4954a956ed66ada10a5/NextcloudTalk/NCRoom.swift#L256-L269
(probably reusable as well)

**Comment by SystemKeeper** [comment]
Cool stuff 👍 

Regarding https://github.com/nextcloud/talk-ios/actions/runs/11442426101/job/31832743043?pr=1845, you need to run `generate-localizable-strings-file.sh` to update the localizable files if you change any translations :)

**Review by SystemKeeper — APPROVED** [review]
Cool, thanks a lot :)

**Review by Ivansss — APPROVED** [review]
Thanks a lot for this PR @rapterjet2004 :) Nice work!

**Pull request merged**